src/Resource/CloudAccount/Resource.php

```diff
@@ -66,4 +68,17 @@ class Resource extends Model {
     'temp_domain',
     'unix_username'
   ];
+
+  /**
+   * Switches PHP version on this cloud account.
+   *
+   * @param string $version Target PHP version
+   * @return Resource $this
+   * @throws ResourceException If endpoint not available
+   * @throws ApiException If request fails
+   */
+  public function setPhpVersion(string $version) : Resource {
+    $this->_getEndpoint()->setPhpVersion($this, $version)->wait();
```

Comment:
Strictly, IMHO, the waits should be optional. a parameter passed in. Let the developer decide whether this is a fire & forget or if they want to wait until it's complete. in the CLI, maybe they are enforced but that's one level up.

Reply:
This is something I was debating over yesterday, and went with it because (a) was causing me too much pain, still having the old value after making the api call, and (b) when invoked from the _Resource_ there's no way to access the _Endpoint_ (which is where `wait()` is).

My solution here is for `$endpoint->action()` to not wait by default*, and for `$resource->proxiedAction()` to always wait by default. I considered a `$wait` parameter, but this seemed like it would become klunky for some actions (especially where some args might be optional).  This seemed, to me, like the best bet for usability.

\* _still respecting the `wait` settings in config_ 

Open to further discussion on this point.

Reply:
Will reassess/address this later per wells